File: hw 3/complex.py
import random
import math
from number import *

class Complex(Number):
    def __init__(self):
        self.real_part = 0
        self.imaginary_part = 0

    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        self.real_part = int(strArray[i])
        self.imaginary_part = int(strArray[i+1])
        i += 2
        return i

    def InRnd(self):
        self.real_part = random.uniform(1, 2000)
        self.imaginary_part = random.uniform(1, 2000)

    def Print(self):
        print("It is complex number: real part = {}, imaginary part = {}. Convert to real number {}".format(self.real_part, self.imaginary_part, self.ConvertToRealNumber()))
        pass

    def Write(self, ostream):
        ostream.write("It is complex number: real part = {}, imaginary part = {}. Convert to real number {}".format(self.real_part, self.imaginary_part, self.ConvertToRealNumber()))
        pass

    def ConvertToRealNumber(self):
        return math.sqrt(self.real_part**2 + self.imaginary_part**2)
        pass

File: hw_2/complex.cpp
//------------------------------------------------------------------------------
// complex.cpp - содержит процедуру ввода параметров
// для уже созданного комплексного числа
//------------------------------------------------------------------------------

#include "complex.h"
#include <fstream>
#include <cmath>
//------------------------------------------------------------------------------

Complex::Complex() : real_part(0), imaginary_part(1) {}

// Ввод комплексного числа из файла
void Complex::In(std::ifstream &ifst) {
    ifst >> this->real_part >> this->imaginary_part;
}

// Случайный ввод комплексного числа
void Complex::InRnd() {
    this->real_part = Random();
    this->imaginary_part = Random();
}

//------------------------------------------------------------------------------
// Вывод  комплексного числа в форматируемый поток
void Complex::Out(std::ofstream &ofst) {
    ofst << "It is complex number: real part = " << real_part
         << ", imaginary part = " << imaginary_part
         <<". Convert to real number " << ConvertToRealNumber() << "\n";
}

//------------------------------------------------------------------------------
// Вычисление действительного представления комплексного числа
double Complex::ConvertToRealNumber() {
    return std::sqrt(real_part * real_part + imaginary_part * imaginary_part);
}
